core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -519,44 +523,43 @@ private[log] class Cleaner(val id: Int,
   private def compressMessages(buffer: ByteBuffer,
                                compressionCodec: CompressionCodec,
                                messageFormatVersion: Byte,
-                               messageAndOffsets: Seq[MessageAndOffset]) {
+                               messageAndOffsets: Seq[MessageAndOffset]): Int = {
     require(compressionCodec != NoCompressionCodec, s"compressionCodec must not be $NoCompressionCodec")
-    if (messageAndOffsets.nonEmpty) {
```

Comment:
Actually you're right. I was probably thinking of a bug I introduced when I pulled the `CleanerStats` updating out of this function 😄